packages/services/api/src/modules/preflight-script/providers/preflight-script.provider.ts

```diff
@@ -0,0 +1,45 @@
+import { Injectable, Scope } from 'graphql-modules';
+import { AuthManager } from '../../auth/providers/auth-manager';
+import { Logger } from '../../shared/providers/logger';
+import { Storage } from '../../shared/providers/storage';
+import type { PreflightScriptModule } from './../__generated__/types';
+
+@Injectable({
+  global: true,
+  scope: Scope.Operation,
+})
+export class PreflightScriptProvider {
+  private logger: Logger;
+
+  constructor(
+    logger: Logger,
+    private storage: Storage,
+    private authManager: AuthManager,
+  ) {
+    this.logger = logger.child({ source: 'PreflightScriptProvider' });
+  }
+
+  getPreflightScript(targetSlug: string) {
+    return this.storage.getPreflightScript({ targetSlug });
+  }
+
+  async createPreflightScript(
+    targetSlug: string,
+    { sourceCode }: PreflightScriptModule.CreatePreflightScriptInput,
+  ) {
+    const currentUser = await this.authManager.getCurrentUser();
+
+    return this.storage.createPreflightScript({
+      createdByUserId: currentUser.id,
+      sourceCode,
```

Comment:
we need to sanitize this input. Can't rely on FE validations on storing this.

Let's maybe run https://github.com/NodeSecure/js-x-ray or something like this tool so we can throw error on the client. Same logic should apply on updates. At a high level I can think of things that we don't want to store. That library has some good rules and we can apply most of those and test what all rules are unnecessary for us.
- anything that touches `Buffer` someone can do a Buffer overflow attack
- No `eval` or anything around the lines of that can cause eval to happen (yes I know we run in a worker but still)
- we don't want to store invalid JS code of user

We need to limit the size of these scripts. Someone can send in a 100mb minified JS just to spam us. Let's add a hard limit of 100KB (seems reasonable to me even minified [jquery](https://bundlephobia.com/package/jquery@3.7.1) is ~80KB) to begin with for these user scripts. If someone has bigger script we should learn first then increase limit.

Reply:
done, limited to 5_000 chars